python/lsst/meas/extensions/ngmix/EMPsfApprox.py

```diff
@@ -0,0 +1,151 @@
+
+    # Constants to identify the failures.  Must match flagDefs below
+    FAILURE = 0
+    RANGE_ERROR = 1
+    MAXITER = 2
+    NO_PSF = 3
+
+    gaussian_pars_len = 6
+
+    FLAGDEFS = [ FlagDefinition("flag", "General failure error"),
+        FlagDefinition("flag_rangeError", "Iteration error in Gaussian parameters."),
+        FlagDefinition("flag_maxIters", "Fitter exceeded maxIters setting without converging."),
+        FlagDefinition("flag_noPsf", "No PSF attached to the exposure.")
+    ]
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.SHAPE_ORDER
+
+    def __init__(self, config, name, schema, metadata):
+        SingleFramePlugin.__init__(self, config, name, schema, metadata)
+
+        self.flagHandler = FlagHandler.addFields(schema, name, FlagDefinitionVector(self.FLAGDEFS))
+        self.iterKey = schema.addField(name + '_iterations', type=int, doc="number of iterations run")
+        self.fdiffKey = schema.addField(name + '_fdiff', type=float, doc="fit difference")
+
+        # Add ShapeletFunction keys for the number of Gaussians requested
+        self.keys = []
+        for i in range(config.nGauss):
+            key = lsst.shapelet.ShapeletFunctionKey.addFields(schema,
+                  "%s_%d"%(name, i), "ngmix EM gaussian", "pixels", "",
+                  0, lsst.shapelet.HERMITE)
+            self.keys.append(key)
+        self.msfKey = lsst.shapelet.MultiShapeletFunctionKey(self.keys)
+
+    def measure(self, measRecord, exposure):
+
+        if exposure.getPsf() is None:
+            raise MeasurementError(self.flagHandler.getDefinition(self.NO_PSF).doc, self.NO_PSF)
+        psfArray = exposure.getPsf().computeKernelImage().getArray()
+        psfObs = ngmix.observation.Observation(psfArray,
+            jacobian=ngmix.UnitJacobian(row=(psfArray.shape[0]-1)/2, col=(psfArray.shape[1]-1)/2))
+
+        #  Need a guess at the sum of the diagonal moments
+        nGauss = self.config.nGauss
+        shape = exposure.getPsf().computeShape()
+        Tguess = shape.getIxx() + shape.getIyy()
+        emPars={'maxiter':self.config.maxIters, 'tol':self.config.tolerance}
+
+        runner=EMRunner(psfObs, Tguess, nGauss, emPars)
+        runner.go(ntry=self.config.nTries)
+        fitter=runner.get_fitter()
+        res=fitter.get_result()
+
+        #   Set the results, including the fit info returned by EMRunner
+        measRecord.set(self.iterKey, res['numiter'])
+        measRecord.set(self.fdiffKey, res['fdiff'])
+
+        #   We only know about two EM errors.  Anything else is thrown as "unknown".
+        if res['flags'] != 0:
+            if res['flags'] & EM_RANGE_ERROR:
+                raise MeasurementError(self.flagHandler.getDefinition(self.RANGE_ERROR).doc, self.RANGE_ERROR)
+            if res['flags'] & EM_MAXITER:
+                raise MeasurementError(self.flagHandler.getDefinition(self.MAXITER).doc, self.MAXITER)
+            raise RuntimeError("Unknown EM fitter exception")
```

Comment:
As with my comment in the other plugin, please check that a RuntimeError is appropriate to be thrown.